### PR TITLE
add ability to specify env vars for spin build in test runner

### PR DESCRIPTION
**Pull request by karthik2804**
Adds the ability to specify environment variables as the `http_python_template_smoke_test` needs a couple.

### CODE REVIEW

File: tests/testcases/mod.rs
Comment (dicej) [inline tests/testcases/mod.rs:342]: I think this needs to be `"{};{}"` on Windows.

Comment (dicej) [reply tests/testcases/mod.rs:342]: The existing expression above your changes (`format!("{path}:{}", spin_binary().parent().unwrap().display())`) should probably also be changed.  Or is this code never intended to run on Windows?

Comment (karthik2804) [reply tests/testcases/mod.rs:342]: I think getting the integration tests to be cross-platform is something on the backlog. @rylev would be the best person for that question.

**Review by dicej — APPROVED** [review]
LGTM.  Thanks for doing this.

File: tests/testcases/mod.rs
Comment (itowlson) [inline tests/testcases/mod.rs:342]: This works today because `build.env("PATH")` has not been modified since it was set to `path`, but feels a bit fragile.  If you want to prepend to the command PATH, it would be good to use the command PATH as the source of truth.  (I don't know if this will ever be an issue in practice though!)

File: tests/testcases/mod.rs
Comment (itowlson) [inline tests/testcases/mod.rs:336]: Consider pulling this out into a named function - it requires close reading of this code to see what's going on.  E.g.

```
let custom_path = env.absolutise_path(value);
```

**Comment by itowlson** [comment]
ONCE AGAIN IVAN CLICKS THE REVIEW BUTTON MERE SECONDS AFTER KARTHIK CLICKS THE MERGE BUTTON.  How do you _do_ that?  _grin_

**Pull request merged**
